packages/common/src/ui/components/inputs/TextInput/PositiveNumberInput.tsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import { NumericTextInput } from '@common/components';
+import { NumUtils } from '@common/utils';
+import { NumericTextInputProps } from './NumericTextInput';
+
+interface NonNegativeNumberProps
+  extends Omit<NumericTextInputProps, 'onChange'> {
+  max?: number;
+  onChange: (newValue: number) => void;
+}
+
+// where NonNegative is n >=0
+export const PositiveNumberInput = React.forwardRef(
```

Comment:
I took what was in the cell version of this and made it an input. Then made the cell use this input